cmd/blueprints.go

```diff
@@ -742,31 +788,64 @@ func bpPull(pullOptions, org, repo, path, outdir string,
 //   Last, run goimport to check for missing or unused import
 //   statements and clean up any code formatting issues
 //
-func bpCreate(rn string) string {
+//func bpCreate(rn string)  bpCreateResponse {
+func bpCreate(rn string) (reply bpCreateResponse) {
 	var filelist []bpLocation
 
 	// Returns a list of all files in the blueprint directory
 	// including there path, .datamgr/Makefile ....
 	inputBlueprintFiles, err := bpRead(bpFile)
-	if err != nil {
-		fmt.Println(err)
-	}
 
 	// Read the definition file
 	defs := bpDef{}
 
 	//TODO: should be a method not a function
 	err = bpReadDefinitions(&defs)
 	if err != nil {
-		return (err.Error())
+		var errorReply bpGenericResponseItem = bpGenericResponseItem{
+			Name:    rn,
+			Content: "Failed reading defintions file",
+		}
+
+		reply.Blueprints = append(reply.Blueprints, errorReply)
+		return (reply)
 	}
 
+	// TODO: move to a function
+	lb := startBlocksSpinner("Loading blocks")
 	bpInputData := bpData{}
 	err = bpDataMapper(defs, &bpInputData)
 
+	var newEndPoint endpointConfig
+	if epList, err := newEndPoint.loadFromDefinitions(defs); err != nil {
```

Comment:
If we want to support other routes in the future, this should be an explicit gorilla section in the definitions file